task1Daniv/stringlist.cpp

```diff
@@ -0,0 +1,250 @@
+        temp[nextNodeIndex] = nullptr;
+    }
+    return  temp;
+}
+
+void push(List &list, const String &string)
+{
+    if(list == nullptr)
+    {
+        return;
+    }
+    String textStr = static_cast<String>(malloc(sizeof(char) * strlen(string) + 1));
+    strcpy(textStr, string);
+
+    if(list[dataIndex] == nullptr)
+    {
+        list[dataIndex] = textStr;
+    }
+    else
+    {
+        auto temp = list;
+        while(list[nextNodeIndex] != nullptr)
+        {
+            list = reinterpret_cast<List>(list[nextNodeIndex]);
+        }
+        List node = static_cast<List>(malloc(sizeof(char*) * 2));
+        node[dataIndex] = textStr;
+        node[nextNodeIndex] = nullptr;
+        list[nextNodeIndex] = reinterpret_cast<String>(node);
+        list = temp;
+    }
+}
+
+void print(const List &list)
+{
+    if(list == nullptr)
+    {
+        return;
+    }
+    auto temp = list;
+    for(int i = 0; i < size(list); ++i)
+    {
+        printf("%s\n",temp[dataIndex]);
+        temp = reinterpret_cast<List>(temp[nextNodeIndex]);
+
+    }
+}
+
+int size(const List &list)
+{
+    if(list == nullptr)
+    {
+        return -1;
+    }
+
+    int totalSize = 0;
+    auto temp = list;
+    while (temp != nullptr && temp[dataIndex] != nullptr)
+    {
+        totalSize++;
+        temp = reinterpret_cast<List>(temp[nextNodeIndex]);
+    }
+    return totalSize;
+}
+
+int index(const List &list, const String &string)
+{
+    if(list == nullptr || string == nullptr)
+    {
+        return -2;
+    }
+
+    auto temp = list;
+
+    for(int i = 0; i != size(list); i++)
+    {
+        if(strcmp(string,temp[dataIndex])==0)
+        {
+            return i;
+        }
+        temp = reinterpret_cast<List>(temp[nextNodeIndex]);
+    }
+    return -1;
+
+}
+
+void destroy(List &list)
+{
+    if(list == nullptr)
+    {
+        return;
+    }
+
+    for(int i = 0; i != size(list); ++i)
+    {
+          List temp = reinterpret_cast<List>(list[nextNodeIndex]);
+          free(list[dataIndex]);
+          free(list);
+          list = temp;
+    }
+
+    list = nullptr;
+}
+
+void remove(List &list, const String &string)
+{
+    if(list == nullptr || string == nullptr)
+    {
+        return;
+    }
+
+    int stringIndex = index(list,string);
+
+    if(stringIndex < 0 )
+    {
+        return;
+    }
+    auto temp = list;
+    auto prev = list;
+    if(stringIndex == 0 && size(list) !=0)
+    {
+        free(list[dataIndex]);
+        list = reinterpret_cast<List>(list[nextNodeIndex]);
+    }
+    else if(stringIndex == 0 && size(list) == 0)
+    {
+        free(list[dataIndex]);
+        free(list);
+        list = nullptr;
+    }
+    else
+    {
+        for(int i = 0; i < stringIndex; ++i)
+        {
+            prev = temp;
+            temp = reinterpret_cast<List>(temp[nextNodeIndex]);
+        }
+
+        prev[nextNodeIndex] = temp[nextNodeIndex];
+        free(temp[dataIndex]);
+        free(temp);
+    }
+}
+
+void sort(List &list)
```

Comment:
Simplify behavior of this function, plz...
Three simultaneous lists with back-relations is too much